### PR TITLE
BundleHandler uses GetChangesMapArgs when available

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following PR updates the BundleHandler code to use `GetChangesMapArgs` from Juju when possible. Additionally also updated the schema to have the latest 2.7-dev schema.

### QA

Ensure you have the following branch checked out https://github.com/juju/juju/pull/10601, or if it's merged, use develop branch.

Deploy the branch so we can test against it:
```sh
juju bootstrap lxd test --no-gui --build-agent
```

To ensure that we're using the right method call, change the following:
```diff
diff --git a/juju/bundle.py b/juju/bundle.py
index f6fd91f..6df6635 100644
--- a/juju/bundle.py
+++ b/juju/bundle.py
@@ -140,7 +140,7 @@ class BundleHandler:
                 bundleurl=entity_id,
                 yaml=yaml.dump(self.bundle))
         else:
-            log.info("bundle_facade using GetChangesMapArgs API to deploy bundle changes")
+            log.critical("bundle_facade using GetChangesMapArgs API to deploy bundle changes")
             self.plan = await self.bundle_facade.GetChangesMapArgs(
                 bundleurl=entity_id,
                 yaml=yaml.dump(self.bundle))
```

Then run:
```sh
tox -e example -- examples/deploy_bundle.py
```

You should see in the output of the example:
```sh
Connecting to model
Deploying bundle
bundle_facade using GetChangesMapArgs API to deploy bundle changes
Waiting for active
Successfully deployed!
Removing bundle
Disconnecting from model
Success
```

### CODE REVIEW

File: juju/client/connection.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
     'ApplicationScaler': {'versions': [1]},
     'Backups': {'versions': [1, 2]},
     'Block': {'versions': [2]},
-    'Bundle': {'versions': [1, 2, 3]},
+    'Bundle': {'versions': [1, 2, 3, 4]},
```
Comment (SimonRichardson) [inline juju/client/connection.py]: Change the bundle facade from 3 to 4.

File: juju/bundle.py [hunk 2]
```diff
@@ -135,9 +135,15 @@ def __init__(self, model, trusted=False, forced=False):
...
-        self.plan = await self.bundle_facade.GetChanges(
-            bundleurl=entity_id,
-            yaml=yaml.dump(self.bundle))
+        if client.BundleFacade.best_facade_version(self.model.connection()) < 4:
+            self.plan = await self.bundle_facade.GetChanges(
+                bundleurl=entity_id,
+                yaml=yaml.dump(self.bundle))
+        else:
+            log.info("bundle_facade using GetChangesMapArgs API to deploy bundle changes")
+            self.plan = await self.bundle_facade.GetChangesMapArgs(
```
Comment (SimonRichardson) [inline juju/bundle.py]: Use the latest method if it's available

**Comment by SimonRichardson** [comment]
https://trello.com/c/NRsXYWNy/1800-update-pylibjuju-27-with-new-facade-changes

File: juju/client/_client.py [hunk 3]
```diff
@@ -3,20 +3,20 @@
 
 from juju.client._definitions import *
 
-from juju.client import _client1, _client2, _client3, _client4, _client5, _client6, _client7, _client8, _client9, _client10, _client12
+from juju.client import _client2, _client1, _client3, _client4, _client5, _client8, _client7, _client9, _client10, _client6, _client12
```
Comment (achilleasa) [inline juju/client/_client.py]: Is the ordering intentional to mess with people that have OCD? ;-)

Comment (SimonRichardson) [reply juju/client/_client.py]: I'm triggered EVERYTIME... I might work out why it does this.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**